src/action.cpp

```diff
@@ -0,0 +1,116 @@
+#include<cmath>
+
+#include "action.h"
+#include "core.h"
+
+Action::Action(Character* actor, double score) {
+    this->actor_ = actor;
+    this->score_ = score;
+}
+
+Action::~Action() {
+}
+
+Character* Action::GetActor() {
+    return this->actor_;
+}
+
+void Action::SetActor(Character* actor) {
+    this->actor_ = actor;
+}
+
+double Action::GetScore() {
+    return this->score_;
+}
+
+void Action::SetScore(double score) {
+    this->score_ = score;
+}
+
+TrivialAction::TrivialAction(Character* actor, double score) : Action(actor, score) {
+}
+
+bool TrivialAction::IsValid(const CVC& gamestate) {
+    return true;
+}
+
+void TrivialAction::TakeEffect(CVC& gamestate) {
+    printf("trivial by %d\n", this->GetActor()->GetId());
+}
+
+AskAction::AskAction(Character *actor, double score, Character *target, double request_amount) : Action(actor, score) {
```

Comment:
I think keeping lines to 80 cols is in most style guides now. I think setting up clang-format integration with vim is worthwhile - I just hit ctrl-k and it formats the current line (or multiple lines if selected). It is quite good at figuring out how to wrap lines well.

Reply:
cool, I'll check it out.

Reply:
I've got another change on a separate branch that'll be difficult to merge if I do this now. so I'll defer this change until then and I'll fix my style after I merge this, on top of that other commit